Paralayout/GeometryAdditions.swift

```diff
@@ -89,31 +89,37 @@ extension CGRect {
     public func inset(left: CGFloat = 0.0, top: CGFloat = 0.0, right: CGFloat = 0.0, bottom: CGFloat = 0.0) -> CGRect {
         return inset(by: UIEdgeInsets(top: top, left: left, bottom: bottom, right: right))
     }
-    
-    /// Outsets the rect, if necessary, to snap to the nearest pixel at the specified scale.
-    /// - parameter scaleFactor: The pixel scale to use, e.g. a UIScreen, UIView, or explicit value (pass `0` to *not*
-    /// snap to pixel).
-    /// - returns: A new rect with pixel-aligned boundaries, enclosing the original rect.
-    public func expandToPixel(_ scaleFactor: ScaleFactorProviding) -> CGRect {
-        return CGRect(
-            left: minX.floorToPixel(in: scaleFactor),
-            top: minY.floorToPixel(in: scaleFactor),
-            right: maxX.ceilToPixel(in: scaleFactor),
-            bottom: maxY.ceilToPixel(in: scaleFactor)
-        )
+
+    /// Insets the rect equally on all sides.
+    /// - parameter inset: The amount by which to contract each edge of the receiver.
+    /// - returns: A new rect with the inset applied.
+    public func inset(allSides inset: CGFloat) -> CGRect {
+        return insetBy(dx: inset, dy: inset)
     }
-    
-    /// Insets the rect, if necessary, to snap to the nearest pixel at the specified scale.
-    /// - parameter scaleFactor: The pixel scale to use, e.g. a UIScreen, UIView, or explicit value (pass `0` to *not*
-    /// snap to pixel).
-    /// - returns: A new rect with pixel-aligned boundaries, enclosed by the original rect.
-    public func contractToPixel(_ scaleFactor: ScaleFactorProviding) -> CGRect {
-        return CGRect(
-            left: minX.ceilToPixel(in: scaleFactor),
-            top: minY.ceilToPixel(in: scaleFactor),
-            right: maxX.floorToPixel(in: scaleFactor),
-            bottom: maxY.floorToPixel(in: scaleFactor)
+
+    /// Insets the rect horizontally and/or vertically.
+    /// - parameter horizontal: The inset to apply to the left and right edges. Defaults to `0`.
+    /// - parameter vertical: The inset to apply to the top and bottom edges. Defaults to `0`.
+    /// - returns: A new rect with the inset(s) applied.
+    public func inset(horizontal dx: CGFloat = 0.0, vertical dy: CGFloat = 0.0) -> CGRect {
```

Comment:
Feels like this is identical to `insetBy(dx:dy:)`, but with defaults. Any reason we shouldn't just use that one? And maybe add versions that default the `dx` or `dy`, like `insetBy(dx:)`.

Reply:
Yeah, I think it would make sense to extend that, except we can't name it `insetBy(dx:dy:)` (with the default values) because you'll get `Ambiguous use of 'insetBy(dx:dy:)'` when you try to use it. We could add two separate methods though: `insetBy(dx:)` and `insetBy(dy:)`. Is that we you're thinking?

Reply:
Actually, I wonder if this is worth including at all... that only ends up saving 7 characters for each method call, while opening up some potential to conflicting with UIKit in a future release if they add default values.

Reply:
Yeah, I was thinking of adding the two individual methods. I'm also totally fine leaving these out. I'm generally of the opinion that little one-line convenience methods like this don't belong in frameworks for exactly the reason you mention.

Reply:
Fair enough. Removed in efb8d00. We can add it back later if needed.